tests/integration_tests/tests/agentless_tests/test_resume.py

```diff
@@ -329,3 +335,44 @@ def test_force_resume_cancelled_nonresumable(self):
                         .runtime_properties['resumed'])
         self.assertTrue(self.client.node_instances.get(instance2.id)
                         .runtime_properties['marked'])
+
+    def test_restart(self):
+        """Test that the restart builtin workflow can be resumed.
+
+        We'll test resuming at both the "stop" stage and the "start" stage.
+        If an operation doesn't run when we expect it to, then the
+        wait_for_log will timeout - the fact that it doesn't, is essentially
+        the assert in this test.
+        """
+        dep = self._create_deployment()
+        execution = self.execute_workflow(
+            workflow_name='restart',
+            wait_for_execution=False,
+            parameters={'node_ids': ['node1']},  # no need for both
+            deployment_id=dep.id)
+
+        # wait for the 'stop' operation to begin, and cancel
+        self._wait_for_log(execution, message=self.wait_message + ' - STOP')
+        self.client.executions.cancel(execution.id)
+        self.wait_for_execution_to_end(execution)
+
+        # now resume...
+        self.client.executions.resume(execution.id)
+        self._create_target_file('/tmp/wait_stop')
```

Comment:
maybe change the name to create_execution_lock or something that will describe the point of this line and related line in the rest of the test.

Reply:
but it's not a lock, and I don't want to add additional naming/abstraction here, I think working on this level is good, makes it easy to understand how does it work.

Added a docstring to the method describing why is it needed

Reply:
I don't like the added handling of '/tmp/wait_stop' file, it requires the reader to get into the test framework to really understand why this line is connected

Reply:
changed it to be runtime_properties instead of files, and a list of "unlocked operations" in runtime props